src/client/Microsoft.Identity.Client/ManagedIdentity/ServiceFabricFederatedManagedIdentitySource.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client.ManagedIdentity
+{
+    internal class ServiceFabricFederatedManagedIdentitySource : AbstractManagedIdentity
```

Comment:
Seems like a lot of code for this. Is there no opportunity for code reuse?

Reply:
There is, but I wanted to make this completely separate. however, it doesn't hurt to merge the two